selftests/unit/test_parser.py

```diff
@@ -14,7 +14,10 @@ def __init__(self):
         super(SilentParser, self).__init__(prog='avocado')
 
     def error(self, message):
-        raise
+        """
+        Don't sys.exit, but only raise RuntimeError
```

Comment:
The argparse on error raises `SystemExit` exception (basically `sys.exit`). By overriding the `error` method we change it to `RuntimeError`. Both can be easily detected/catched, but the `SystemExit` could mean anything, while our `RuntimeError` with custom message is easier to detect.

Second reason is, that the default `error` handler writes directly to `sys.stderr`, which pollutes the output of `unittest`.